src/Microsoft.AspNetCore.Authentication/ClaimsTransformationMiddleware.cs

```diff
@@ -39,13 +39,18 @@ public class ClaimsTransformationMiddleware
             handler.RegisterAuthenticationHandler(context.GetAuthentication());
             try
             {
-                if (Options.Transformer != null)
+                var transform = Options.Transformer;
+                if (transform == null && Options.TransformerType != null)
+                {
+                    transform = context.RequestServices.GetService(Options.TransformerType) as IClaimsTransformer;
```

Comment:
> If we add a check that throws if neither transform is specified, that most certainly is a functional breaking change...

Except that's not what I'm suggesting: https://github.com/aspnet/Security/pull/1014#discussion_r84986523 :smile: 

If you `UseClaimsTransformation()` "with nothing else", this code path will never be reached (because `TransformerType` will be null).